src/AppBundle/Tests/Entity/PublicationTest.php

```diff
@@ -14,16 +15,73 @@ public function testAddGenre() {
         $publication = new Book();
         $genre = new Genre();
         $publication->addGenre($genre);
-        $publication->addGenre($genre);
+        //$publication->addGenre($genre);
         $this->assertEquals(1, count($publication->getGenres()));
     }
     
     public function testAddContribution() {
         $publication = new Compilation();
         $genre = new Contribution();
         $publication->addContribution($genre);
-        $publication->addContribution($genre);
+        //$publication->addContribution($genre);
         $this->assertEquals(1, count($publication->getContributions()));
     }
+    public function testAppendNote(){
+        
+        $testNote = "This is a note to append.";
+        
+        $publication1 = new Book();
+        $publication2 = new Compilation();
+        $publication3 = new Periodical();
+        
+        $publication1->appendNote($testNote);
+        $publication2->appendNote($testNote);
+        $publication3->appendNote($testNote);
+        
+        $this->assertEquals($testNote, $publication1->getNotes());
+        $this->assertEquals($testNote, $publication2->getNotes());
+        $this->assertEquals($testNote, $publication3->getNotes());
+    }
```

Comment:
These should be in three different tests.

As an example, if only `$this->assertEquals($testNote, $publication3->getNotes());` fails, then phpunit will report the entire test as failed, even though the other two actually passed.

Reply:
Solved!